hermes/crates/cbork/cddl-parser/src/lib.rs

```diff
@@ -7,20 +7,53 @@
 use std::fmt::Debug;
 
 pub use pest::Parser;
-use pest_derive::Parser;
 
 extern crate derive_more;
 use derive_more::{Display, From};
+use pest::error::Error;
 
-// Parser with DEBUG rules.  These rules are only used in tests.
-#[derive(Parser)]
-#[grammar = "grammar/cddl.pest"]
-#[grammar = "grammar/cddl_test.pest"] // Ideally this would only be used in tests.
-pub struct CDDLParser;
+pub mod rfc_8610 {
+    pub use pest::Parser;
+
+    #[derive(pest_derive::Parser)]
+    #[grammar = "grammar/cddl.pest"]
+    pub struct RFC8610Parser;
+}
+
+pub mod cddl_test {
+    pub use pest::Parser;
+
+    // Parser with DEBUG rules. These rules are only used in tests.
+    #[derive(pest_derive::Parser)]
+    #[grammar = "grammar/cddl.pest"]
+    #[grammar = "grammar/cddl_test.pest"] // Ideally this would only be used in tests.
+    pub struct CDDLTestParser;
+}
+
+/// Represents different parser extensions for handling CDDL specifications.
+pub enum Extension {
+    /// RFC8610 ONLY limited parser.
+    RFC8610Parser,
+    /// RFC8610 and RFC9615 limited parser.
+    RFC9615Parser,
+    /// RFC8610, RFC9615, and CDDL modules.
+    CDDLParser,
+    /// Same as CDDLParser but includes the `cddl_test.pest` file for integration test
+    /// usage, mainly for development testing.
+    CDDLTestParser,
```

Comment:
Because we can call the `cddl_test` parser directly in the unit tests, we can exclude it from this enum, and also the error type.  That way the public API only has the valid parsers, not out test parser which is only used for the Unit tests.